src/client/Microsoft.Identity.Client.Brokers/RuntimeBroker.cs

```diff
@@ -0,0 +1,392 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Cache;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Instance.Discovery;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Platforms.Features.DesktopOs;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Broker
+{
+    // TODO: wpf crashes on the second ATI
+    // TODO: need to map exceptions 
+    //   - WAM's retrayble exception?
+    //   - cancellation exception for when the user closes the browser
+
+    // TODO: bug around double interactive auth https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1858419 - block users from calling ATI twice with a semaphore (Undo the fix)
+    // TODO: call start-up only once (i.e. initialize core object only once) 
+    // TODO: pass in claims - try {"access_token":{"deviceid":{"essential":true}}} (Clarify)
+
+    // TODO: multi-cloud support (blocked by WAM bug)
+    // TODO: add logging (Blocked - a C++ API exists, no C# API yet as it's pretty complex, waiting for msalruntime to expose it)
+
+    internal class RuntimeBroker : IBroker
+    {
+        private readonly ICoreLogger _logger;
+        private readonly IntPtr _parentHandle = IntPtr.Zero;
+        internal const string ErrorMessageSuffix = " For more details see https://aka.ms/msal-net-wam";
+        private readonly WindowsBrokerOptions _wamOptions;
+        private const string WamErrorPrefix = "WAM Error ";
+
+        /// <summary>
+        /// Ctor. Only call if on Win10, otherwise a TypeLoadException occurs. See DesktopOsHelper.IsWin10
+        /// </summary>
+        public RuntimeBroker(
+            CoreUIParent uiParent,
+            ApplicationConfiguration appConfig,
+            ICoreLogger logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            _parentHandle = GetParentWindow(uiParent);
+
+            _wamOptions = appConfig.WindowsBrokerOptions ??
+                WindowsBrokerOptions.CreateDefault();
+
+            if (_wamOptions.ListWindowsWorkAndSchoolAccounts)
+            {
+                throw new NotImplementedException("The new broker implementation does not yet support Windows account discovery (ListWindowsWorkAndSchoolAccounts option)");
+            }
+        }
+
+        /// <summary>
+        /// Acquire Token Interactively 
```

Comment:
I recommend deleting the comments on internal methods, unless the method is doing something "out of the ordinary" that needs to be documented. The interface should have some comments.

Reply:
done